python/mxnet/initializer.py

```diff
@@ -530,9 +530,9 @@ def _init_weight(self, _, arr):
         nout = arr.shape[0]
         nin = np.prod(arr.shape[1:])
         if self.rand_type == "uniform":
-            tmp = np.random.uniform(-1.0, 1.0, (nout, nin))
+            tmp = random.uniform(-1.0, 1.0, shape=(nout, nin)).asnumpy()
```

Comment:
are these asnumpys necessary?

Reply:
Numpy's SVD is applied to `tmp` array, so the conversions are needed. Currently mxnet seems not have SVD. There is a simple workaround to perform SVD using eigen decomposition (so using `syevd`), however it is not so efficient.